Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -325,6 +325,12 @@ extension BrowserViewController: WKNavigationDelegate {
         self.tabManager.selectedTab?.alertShownCount = 0
         self.tabManager.selectedTab?.blockAllAlerts = false
       }
+      
+      // Mixed-Content validation
+      if let target = navigationAction.targetFrame?.request.url,
+         url.scheme == "http", target.scheme == "https" {
```

Comment:
FYI i think there are other cases that aren't captured by this. for instance `ws://` on an HTTPS page is considered mixed content. and probably `http://localhost` on HTTPS is considered secure.